### DISCUSSION TITLE
verbosity

### DISCUSSION THREAD

**Discussion by xhudik** [com-lihaoyi/mill#1168] [General]
Hi, is there any way how to increase/decrease verbosity? Or, having full logs would help as well

thx, Tomas

**Comment by lefou** [comment]
Mill does not use a logging system like slf4j, but you can increase the verbosity a little bit with `--debug` option. It's not much more info, though.

Do you have a particular issue you want to analyze or is it a general question?

**Comment by xhudik** [comment]
Ho @lefou , thanks for the fast answer!

it is a general question.
I have more questions - not sure where I can ask them...

* command `mill HelloWorld.scalaVersion` gives no result - why?
* any possibility to  chain tasks, e.g.`mill HelloWorld.clean HelloWorld.compile` doesnt work
* `mill HelloWorld.compile` - what exact command is running?
* where mill stores artifacts? For maven - it s clear `~/m2`; gradle is  in mist - `.gradle/caches/modules-2/files-2.1` and somehow `.m2`

sorry for many questions - maybe there is a good documentation that can help - but i didnt find it

**Reply by lefou** [reply]
Yeah, there is a good documentation that should cover most of your questions on the first page: http://www.lihaoyi.com/mill/

I though, that link should be very easy to find... Maybe you can give us a hint where you expected it but didn't find it.

----

FTR, here are the quick answers to your questions.

>     * command `mill HelloWorld.scalaVersion` gives no result - why?

The result is there, but not printed by default. Try `mill show HelloWorld.scalaVersion`

>     * any possibility to  chain tasks, e.g.`mill HelloWorld.clean HelloWorld.compile` doesnt work

`mill all HelloWorld.clean HelloWorld.compile`

or

`mill "HelloWorld.{clean,compile}"`

>     * `mill HelloWorld.compile` - what exact command is running?

`mill plan HelloWorld.compile`

>     * where mill stores artifacts? For maven - it s clear `~/m2`; gradle is  in mist - `.gradle/caches/modules-2/files-2.1` and somehow `.m2`

Downloaded artifacts go typically to `~/.cache/coursier`.

Locally published artifacts will go to `.ivy2/local`

**Reply by xhudik** [reply]
very nice!!! 
sure i briefly read http://www.lihaoyi.com/mill/ - but I didnt get those http://www.lihaoyi.com/mill/#command-line-tools Now it is clear to me

Thinking loud - based on my case (read documentation briefly :)) and working with command line help mainly - wouldn't be wise to put them to `mill --help` (or put there some small notion)? Is there any reason why not to put them there?

**Reply by lefou** [reply]
Those are regular targets which can be found with `mill resolve _`. More help can be shown with e.g. `mill describe show`. Adding this into the help may quickly flood the help screen and make it even less useful. Any I think you may agree, nobody reads long pages ;-).

**Reply by xhudik** [reply]
ok, makes sense

btw `mill describe show` - doesnt work, it seems like `describe` is not a valid command

**Reply by lefou** [reply]
Sorry, I meant `inspect`.
